packages/run-protocol/src/runStake/runStake.js

```diff
@@ -0,0 +1,125 @@
+// @ts-check
+import { handleParamGovernance } from '@agoric/governance';
+import { E, Far } from '@endo/far';
+import { makeAttestationFacets } from './attestation.js';
+import { makeRunStakeParamManager } from './params.js';
+import { makeRUNstakeKit } from './runStakeKit.js';
+import { makeRunStakeManager } from './runStakeManager';
+
+/**
+ * @param { ZCF<RunStakeTerms> } zcf
+ * @param {{
+ *   feeMintAccess: FeeMintAccess,
+ *   initialPoserInvitation: Invitation,
+ *   lienBridge: ERef<StakingAuthority>,
+ * }} privateArgs
+ *
+ * @typedef {{
+ *   timerService: TimerService,
+ *   chargingPeriod: bigint,
+ *   recordingPeriod: bigint,
+ *   lienAttestationName: string,
+ *   electionManager: VoteOnParamChange,
+ *   main: {
+ *     MintingRatio: ParamRecord<'ratio'>,
+ *     InterestRate: ParamRecord<'ratio'>,
+ *     LoanFee: ParamRecord<'ratio'>,
+ *     Electorate: ParamRecord<'invitation'>,
+ *   },
+ * }} RunStakeTerms
+ */
+const start = async (
+  zcf,
+  { feeMintAccess, initialPoserInvitation, lienBridge },
+) => {
+  const {
+    main: initialValue,
+    brands: { Stake: stakeBrand },
+    timerService,
+    chargingPeriod,
+    recordingPeriod,
+    lienAttestationName = 'BldLienAtt',
+  } = zcf.getTerms();
+  assert.typeof(chargingPeriod, 'bigint', 'chargingPeriod must be a bigint');
+  assert.typeof(recordingPeriod, 'bigint', 'recordingPeriod must be a bigint');
+
+  const att = await makeAttestationFacets(
+    zcf,
+    stakeBrand,
+    lienAttestationName,
+    lienBridge,
+  );
+  // TODO: remove publicFacet wart
+  const attestBrand = await E(att.publicFacet).getBrand();
+
+  const paramManager = await makeRunStakeParamManager(
+    zcf.getZoeService(),
+    {
+      mintingRatio: initialValue.MintingRatio.value,
+      interestRate: initialValue.InterestRate.value,
+      loanFee: initialValue.LoanFee.value,
+    },
+    initialPoserInvitation,
+  );
+
+  const { wrapPublicFacet, wrapCreatorFacet } = handleParamGovernance(
+    zcf,
+    paramManager,
+  );
+
+  const { zcfSeat: rewardPoolSeat } = zcf.makeEmptySeatKit();
+
+  // assertElectorateMatches(paramManager, otherGovernedTerms);
+
+  /**
+   * We provide an easy way for the policy to add rewards to
+   * the rewardPoolSeat, without directly exposing the rewardPoolSeat to them.
+   *
+   * @type {ReallocateWithFee}
+   */
+  const reallocateWithFee = (fee, fromSeat, otherSeat = undefined) => {
```

Comment:
check out https://github.com/Agoric/agoric-sdk/pull/4877/files#diff-0b7d727a28b84089ac4ebe9b76f8957bb71bfee261640828bad01c4189cbe73fR89

Reply:
my eyes are getting blurry. I'm struggling to see the motivation or something.

Reply:
sorry, I considered being more specific and should have been. The link shows an in-flight PR where `reallocateWithFee` has been expanded in scope to `mintAndReallocate`. I thought you might want to consider the same for parity.

Reply:
Ugh. I'm struggling to get this change right.